packages/tools/fetch-tool/src/fluidFetchSharePoint.ts

```diff
@@ -56,14 +56,11 @@ export async function resolveWrapper<T>(
         }
         return result;
     } catch (e) {
-        if (e.requestResultError) {
-            const parsedBody = JSON.parse(e.requestResult.data);
-            if (parsedBody.error === "invalid_grant"
-                && parsedBody.suberror === "consent_required"
-                && !forceTokenReauth
-            ) {
+        if (e.requestResult) {
+            const parsedBody = e.requestResult.data;
```

Comment:
At a minimum, I'd export proper interface from odsp driver for this code to leverage, and thus be in sync with any changes happening in the driver.

I did not spend that much time examining code, but it feels like createErrorFromResponse() and code around it creates a parallel error handling system, or rather leaks particular driver-specific way of dealing with error handling to outside world.
I'd try to either map these errors to DriverErrorType (generic), OdspErrorType (odsp specific) or maybe even create new enum, but have mapping of various interesting cases into a flat list of known and expected error codes.

Curious, why shape of errors and how we treat them is different on this code path compared to shipping code paths (see createOdspNetworkError())?

Reply:
1.) createErrorFromResponse() is used to create/throw errors in odsp utils package. That error is getting parsed here for doing reauth. The other errors are just thrown off again.
2.) The error that is coming here is not either the DriverErrorType or OdspErrorType but it is the RequestResultError type which is the error type used in the get/put request in odsp utils package to do some auths etc outside of driver code.

Reply:
I think we need more type safety. I'd do at least minimal part here by clearly articulating expected interface for errors.
All the other points you bring can be easily addressed as well (by moving code around from drive to utils package, or just defining new enum and simply using same structure for error reporting, i.e. we can leave driver as is and just make sure that there is some reasonable translation of errors happens in utils package that formats errors in the same way as the driver).

Reply:
Added odspNetworkError to odsp utils and removed parallel createErrorFromResponse. Also added proper parsing of auth error.

Reply:
Added the enums and types here only as the utils cannot depend on driver utils